modules/solid_mechanics/include/timeintegrators/CentralDifferenceDirect.h

```diff
@@ -0,0 +1,54 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ExplicitTimeIntegrator.h"
+
+// Forward declarations
+namespace libMesh
+{
+template <typename T>
+class SparseMatrix;
+}
+class MooseError;
```

Comment:
Since you actually call the function here in the header you need the definition of `mooseError` available so you actually need to `#include "MooseError.h"`. However, you're "getting lucky" because it's already indirectly included via `#include "ExplicitTimeIntegrator.h"`. I would remove this line or alternatively, move your `computeADTimeDerivatives` implementation to the .C file

Reply:
[ab8b9fe](https://github.com/idaholab/moose/pull/28175/commits/ab8b9fe85692cb185dda7f6264efbf1c75a0e597)

I just added the include statement and removed the forward declaration